### DISCUSSION TITLE
The library file semantics.json of library H5P.CoursePresentation-1.22 does not exist.

### DISCUSSION THREAD

**Discussion by Progressive-Programmer** [Lumieducation/H5P-Nodejs-library#2142] [Q&A]
<img width="1440" alt="Screenshot 2022-01-19 at 8 21 23 PM" src="https://user-images.githubusercontent.com/25490780/150154657-0dde5168-9808-4bad-a1d9-9b53bd8fd4db.png">

The file that was being uploaded was taken from h5p.org export option. 
The same file was successfully uploaded in sample setup. 
Frontend - React , Node version v14.17.5
Backend - Node, Node Version v16.13.0

The edit also doesn't load many a times even when the edit API response succeed 
<img width="1440" alt="Screenshot 2022-01-19 at 8 32 42 PM" src="https://user-images.githubusercontent.com/25490780/150156760-6a76fe1c-9ace-49aa-b514-6cf26c716196.png">

**Comment by sr258** [comment]
Does the editor not load in general or only for the course presentation?

**Comment by Progressive-Programmer** [comment]
So let me share the series of events that led to this. 

- The H5P editor load issue has been consistent, it used to vanish once in a while. 
- We raised a ticket earlier and you suggested to upgrade to Node 16 backend+ frontend and clear cache whenever this error happened. 
- After that it worked fine for a while. 
- A new issue popped up when we attempted to upload h5p files. (we were creating h5p files in h5p.org and downloading .h5p exports)
- In one of our other app instance this upload (.json format is not acceptable) was resolved when we cleaned the existing h5p files in the backend and tried uploading them again. 
- This is another instance in which we were trying to upload new files. 
- Upload wasn't successful, so we tried removing existing uploaded files from the server. 
- Once we removed then, we saw this editor issue. The editor was no more loading. 
Now the editor is not loading in our live server. 
The editor does load in local dev environment, but the upload still doesn't work.

**Comment by sr258** [comment]
What library storage backend do you use? File or MongoDB?

**Comment by Progressive-Programmer** [comment]
File

**Comment by sr258** [comment]
Then you should try moving the `libraries/H5P.CoursePresentation-1.22` directory to a temporary directory ("deleting it"). It looks like the storage is corrupt.

**Comment by Progressive-Programmer** [comment]
Should we delete it manually and add it from the repo again ??

**Comment by Progressive-Programmer** [comment]
<img width="1440" alt="Screenshot 2022-01-20 at 2 06 20 PM" src="https://user-images.githubusercontent.com/25490780/150302391-a6ba469f-e2ef-4a7a-8281-0bba37806ea2.png">
You mean this folder, right !

**Comment by sr258** [comment]
Yes, it's this directory. Do you have the libraries directory under source control? If so, it should be enough to revert this directory to the state under source control.

It might also be some caching issue, so wiping the Redis cache could work.

**Comment by Progressive-Programmer** [comment]
Let me try this, allow me some time. I'll update as soon as it's done. Thanks @sr258 Sebastian

**Comment by Progressive-Programmer** [comment]
<img width="1440" alt="Screenshot 2022-01-24 at 5 23 43 PM" src="https://user-images.githubusercontent.com/25490780/150778189-0dbe48d9-1805-40d3-bfb2-b9955f94613a.png">

 "@lumieducation/h5p-react": "^7.5.2",
 "@lumieducation/h5p-server": "^7.5.2",
 We are getting errors for H5PEditor not found
 
 Note: we cleared cache and removed libraries several times. 
 It's not working in local as well

**Comment by sr258** [comment]
I don't believe the error of the last screenshot is related to the first problem. The error in the last screenshot indicates that the JS files for the core editor aren't loaded. Check the network tab to see if files aren't loaded correctly.

Edit: Addition: Check that you've copied the H5P core files into the /h5p directory (the ones that are downloaded from GitHub from h5p-editor-php-library and h5p-php-library)

**Comment by Progressive-Programmer** [comment]
Our backend is NodeJS, do we still require to use these packages ?

**Comment by sr258** [comment]
You only need the JS and CSS parts and image resources, the PHP files can be deleted. But you have to download these files as they constitute the H5P core. You must have done so already at some point, otherwise you wouldn't have had a working H5P installation at all. The script to do this is here: https://github.com/Lumieducation/H5P-Nodejs-library/blob/master/packages/h5p-rest-example-server/download-core.sh
